tests/run/posonly.pyx

```diff
@@ -0,0 +1,323 @@
+#    def f(a = 5, b, /, c): pass
+#    def f(a = 5, b, /): pass
+#    def f(*args, /): pass
+#    def f(*args, a, /): pass
+#    def f(**kwargs, /): pass
+#    def f(/, a = 1): pass
+#    def f(/, a): pass
+#    def f(/): pass
+#    def f(*, a, /): pass
+#    def f(*, /, a): pass
+#    def f(a, /, a): pass
+#    def f(a, /, *, a): pass
+#    def f(a, b/2, c): pass
+
+def test_optional_posonly_args1(a, b=10, /, c=100):
+    """
+    >>> test_optional_posonly_args1(1, 2, 3)
+    6
+    >>> test_optional_posonly_args1(1, 2, c=3)
+    6
+    >>> test_optional_posonly_args1(1, b=2, c=3)
+    Traceback (most recent call last):
+    TypeError: test_optional_posonly_args1() got an unexpected keyword argument 'b'
+    >>> test_optional_posonly_args1(1, 2)
+    103
+    >>> test_optional_posonly_args1(1, b=2)
+    Traceback (most recent call last):
+    TypeError: test_optional_posonly_args1() got an unexpected keyword argument 'b'
+    """
+    return a + b + c
+
+def test_optional_posonly_args2(a=1, b=10, /, c=100):
+    """
+    >>> test_optional_posonly_args2(1, 2, 3)
+    6
+    >>> test_optional_posonly_args2(1, 2, c=3)
+    6
+    >>> test_optional_posonly_args2(1, b=2, c=3)
+    Traceback (most recent call last):
+    TypeError: test_optional_posonly_args2() got an unexpected keyword argument 'b'
+    >>> test_optional_posonly_args2(1, 2)
+    103
+    >>> test_optional_posonly_args2(1, b=2)
+    Traceback (most recent call last):
+    TypeError: test_optional_posonly_args2() got an unexpected keyword argument 'b'
+    >>> test_optional_posonly_args2(1, c=2)
+    13
+    """
+    return a + b + c
+
+# TODO: remove the test below?  would need to hard-code the function with > 255 posonly args
+#def test_syntax_for_many_positional_only():
+#    # more than 255 positional only arguments, should compile ok
+#    fundef = "def f(%s, /):\n  pass\n" % ', '.join('i%d' % i for i in range(300))
+#    compile(fundef, "<test>", "single")
+
+# TODO: remove the test below?  doesn't seem relevant to Cython implementation
+#def test_pos_only_definition(self):
+#    def f(a, b, c, /, d, e=1, *, f, g=2):
+#        pass
+#
+#    self.assertEqual(2, f.__code__.co_argcount)  # 2 "standard args"
+#    self.assertEqual(3, f.__code__.co_posonlyargcount)
+#    self.assertEqual((1,), f.__defaults__)
+#
+#    def f(a, b, c=1, /, d=2, e=3, *, f, g=4):
+#        pass
+#
+#    self.assertEqual(2, f.__code__.co_argcount)  # 2 "standard args"
+#    self.assertEqual(3, f.__code__.co_posonlyargcount)
+#    self.assertEqual((1, 2, 3), f.__defaults__)
+
+def test_pos_only_call_via_unpacking(a, b, /):
+    """
+    >>> test_pos_only_call_via_unpacking(*[1,2])
+    3
+    """
+    return a + b
+
+def test_use_positional_as_keyword1(a, /):
+    """
+    >>> test_use_positional_as_keyword1(a=1)
+    Traceback (most recent call last):
+    TypeError: test_use_positional_as_keyword1() takes no keyword arguments
+    """
+    pass
+
+def test_use_positional_as_keyword2(a, /, b):
```

Comment:
I think it would be good to also always have a correct call to these functions, just to make sure that there is one and they don't accidentally reject every attempt to call them.

Reply:
Done